backend/dataall/modules/mlstudio/services/services.py

```diff
@@ -0,0 +1,207 @@
+"""
+A service layer for ml studio
+Central part for working with ml studio
+"""
+import dataclasses
+import logging
+from dataclasses import dataclass, field
+from typing import List, Dict
+
+from dataall.api.Objects.Stack import stack_helper
+from dataall.core.context import get_context as context
+from dataall.core.environment.db.repositories import EnvironmentParameterRepository
+from dataall.db.api import (
+    ResourcePolicy,
+    Environment, KeyValueTag, Stack,
+)
+from dataall.db import models, exceptions
+from dataall.modules.mlstudio.aws.sagemaker_studio_client import sagemaker_studio_client, SagemakerStudioClient
+from dataall.modules.mlstudio.db.repositories import SageMakerStudioRepository
+
+from dataall.utils.slugify import slugify
+from dataall.modules.mlstudio.db.models import SagemakerStudioUser
+from dataall.modules.mlstudio.services.permissions import (
+    MANAGE_SGMSTUDIO_USERS,
+    CREATE_SGMSTUDIO_USER,
+    SGMSTUDIO_USER_ALL,
+    GET_SGMSTUDIO_USER,
+    SGMSTUDIO_USER_URL,
+    DELETE_SGMSTUDIO_NOTEBOOK,
+)
+from dataall.core.permission_checker import has_resource_permission, has_tenant_permission, has_group_permission
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class SagemakerStudioCreationRequest:
+    """A request dataclass for ml studio user profile creation. Adds default values for missed parameters"""
+    label: str
+    SamlAdminGroupName: str
+    environment: Dict = field(default_factory=dict)
+    description: str = "No description provided"
+    tags: List[str] = field(default_factory=list)
+
+    @classmethod
+    def from_dict(cls, env):
+        """Copies only required fields from the dictionary and creates an instance of class"""
+        fields = set([f.name for f in dataclasses.fields(cls)])
+        return cls(**{
+            k: v for k, v in env.items()
+            if k in fields
+        })
+
+
+def _session():
+    return context().db_engine.scoped_session()
+
+
+class SagemakerStudioService:
+    """
+    Encapsulate the logic of interactions with sagemaker ml studio.
+    """
+    @staticmethod
+    @has_tenant_permission(MANAGE_SGMSTUDIO_USERS)
+    @has_resource_permission(CREATE_SGMSTUDIO_USER)
+    @has_group_permission(CREATE_SGMSTUDIO_USER)
+    # TODO: question, why the * here?
```

Comment:
It was made to force passing `uri` as a kwarg parameter. `uri` is expected for `has_resource_permission` and it's the only way to highlight it somehow. Though it's not necessary

Reply:
I see, having first the arguments we are forcing the uri to be a keyword argument. Thanks!